### PR TITLE
Updated dependencies to latest versions.

### LINKED ISSUE #118: Incompatible with quartz 2.2.3
Quartz 2.2.3 `org.quartz.spi.JobStore` has slightly different method signatures compared to `com.novemberain.quartz.mongodb.MongoDBJobStore`

### PR DESCRIPTION
**Pull request by dorogush**
Fixes #118

### CODE REVIEW

File: src/test/groovy/it/MongoDBJobStoreTest.groovy [hunk 1]
```diff
@@ -6,15 +6,17 @@ import org.bson.Document
...
 import org.quartz.JobDetail
+import org.quartz.JobExecutionContext
+import org.quartz.JobExecutionException
 import org.quartz.JobKey
 import org.quartz.SimpleScheduleBuilder
 import org.quartz.TimeOfDay
 import org.quartz.TriggerBuilder
 import org.quartz.TriggerKey
 import org.quartz.impl.matchers.GroupMatcher
-import org.quartz.jobs.NoOpJob
```
Comment (dorogush) [inline src/test/groovy/it/MongoDBJobStoreTest.groovy]: `NoOpJob` has been moved to `quartz-jobs` artifact. Not worth depending on it.

Comment (michaelklishin) [inline src/test/groovy/it/MongoDBJobStoreTest.groovy]: Fair enough.

File: src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java [hunk 2]
```diff
@@ -32,22 +32,29 @@ public JobCompleteHandler(TriggerAndJobPersister persister, SchedulerSignaler si
...
         log.debug("Trigger completed {}", trigger.getKey());
 
         if (job.isPersistJobDataAfterExecution()) {
             if (job.getJobDataMap().isDirty()) {
                 log.debug("Job data map dirty, will store {}", job.getKey());
-                jobDao.storeJobInMongo(job, true);
+                try {
+                    jobDao.storeJobInMongo(job, true);
+                } catch (JobPersistenceException e) {
+                    throw new RuntimeException(e);
```
Comment (michaelklishin) [inline src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java]: @pwojnowski are you OK with this change? @dorogush may I ask what's the idea here?

File: src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java [hunk 3]
```diff
@@ -32,22 +32,29 @@ public JobCompleteHandler(TriggerAndJobPersister persister, SchedulerSignaler si
...
             }
         }
 
         if (job.isConcurrentExectionDisallowed()) {
             locksDao.unlockJob(job);
         }
 
-        process(trigger, executionInstruction);
+        try {
+            process(trigger, executionInstruction);
```
Comment (michaelklishin) [inline src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java]: Same as above.

File: src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java (hunk 2 above)
Comment (dorogush) [inline src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java]: Quartz [no longer allows](https://github.com/quartz-scheduler/quartz/commit/81fc8672249f889e57a738fd1523fd62c71de09f#diff-ea29ec1c8889f71405d9e9d550d9b6a7R611) throwing `JobPersistenceException` from `JobStore.triggeredJobComplete`

Comment (michaelklishin) [inline src/main/java/com/novemberain/quartz/mongodb/JobCompleteHandler.java]: Ah, so it's a core interface change. Makes sense then.

**Pull request merged**